docsite/docs/Docs/fal-cli/structured-hooks.md

```diff
@@ -0,0 +1,56 @@
+# Passing parametrized data around hooks
+
+Aside from the context around the model that it is bound to, each Fal
+hook (hooks defined under `pre-hook` and `post-hook`) can have a set
+of parameters that would allow it to be parametrized in a per-usage
+basis (rather than having multiple duplicate scripts with statically
+embedded configuration values). This customization can be applied by
+using the `with:` section, for example like the `send_alert.py` below:
+
+```yml
+models:
+  - name: normalize_balances
+    meta:
+      fal:
+        post-hook:
+          - my_regular_hook.py
+          - path: send_alert.py
+            with:
+              channel: "#finance-alerts"
+              severity: 1
+
+   - name: revoke_account_permissions
+     meta:
+       fal:
+         post-hook:
+           - path: send_alert.py
+             with:
+               channel: "#hr-alerts"
+               severity: 0
+```
+
+Each hook can either just specify a path (like `- my_regular_hook.py`) in an unstructured form,
+or specify the full definition of the hook itself by defining `path` and `with` attributes.
```

Comment:
```suggestion
A hook is a dictionary with a `path` and `with` properties. If a string is set 
instead of the dictionary format, it is assumed as the `path` property.
```